health/micro-ui/web/micro-ui-internals/packages/modules/campaign-manager/src/pages/employee/NewCampaignCreate/AppFeatures.js

```
@@ -152,9 +180,9 @@ const AppFeatures = () => {
 
   // Auto-select features when data is loaded
   useEffect(() => {
-    if (!selectedFeaturesByModule && selectedFeatureConfigs?.length > 0 && moduleToggleData?.length) {
+    if (!selectedFeaturesByModule && selectedFeatureConfigs && moduleToggleData?.length) {
       const defaultSelection = moduleToggleData.reduce((acc, moduleCode) => {
-        acc[moduleCode] = [...selectedFeatureConfigs];
+        acc[moduleCode] = [...selectedFeatureConfigs?.[moduleCode]];
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)